pkg/rootlessport/rootlessport_linux.go

```diff
@@ -218,7 +218,13 @@ outer:
 
 	// we only need to have a socket to reload ports when we run under rootless cni
 	if cfg.RootlessCNI {
-		socket, err := net.Listen("unix", filepath.Join(socketDir, cfg.ContainerID))
+		// workaround to bypass the 108 char socket path limit
+		// open the fd and use the path to the fd as bind argument
+		fd, err := unix.Open(socketDir, unix.O_PATH, 0)
```

Comment:
Do we need a deferred close on the FD?

Reply:
yes, I added the close after the listen call, no need to defer this